Formula/r/recc.rb

```diff
@@ -0,0 +1,181 @@
+      -DRECC_CONFIG_PREFIX_DIR=#{etc}
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *buildbox_cmake_args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    %w[cc c++ gcc g++ clang clang++].each do |compiler|
+      (bin/"recc-#{compiler}").write <<~EOS
+        #!/usr/bin/env sh
+
+        #{bin}/recc $(command -v #{compiler}) "$@"
+      EOS
+    end
+
+    # Generate recc-server start helper script
+    (bin/"recc-server").write <<~EOS
+      #!/usr/bin/env sh
+
+      # Get current soft limit for open files
+      current_limit=$(ulimit -Sn)
+
+      REQUIRED_FILES_LIMIT=384
+      # Check if the current limit is less than REQUIRED_FILES_LIMIT
+      if [ "$current_limit" -lt ${REQUIRED_FILES_LIMIT} ]; then
+        ulimit -Sn ${REQUIRED_FILES_LIMIT}
+      fi
+
+      # Run buildbox-casd as recc-server
+      #{bin}/buildbox-casd \
+        --local-server-instance=recc-server \
+        ${1:-#{var}/recc/casd}
+    EOS
+
+    # Generate recc.conf
+    unless (etc/"recc.conf").exist?
+      (etc/"recc.conf").write <<~EOS
+        ##
+        # For the list of configuration parameters please visit:
+        #   https://buildgrid.gitlab.io/recc/configuration-variables.html
+        #
+        # The configuration file settings use lowercase letters without the RECC_ prefix
+        # By default recc reads the configuration options from the following places,
+        # applying settings bottom-up, with 1 being the last applied configuration.
+        # If an option is set in multiple places, the one higher on this list will be the
+        # effective one
+        #
+        # 1. Environment variables
+        # 2. ${cwd}/recc/recc.conf
+        # 3. {$RECC_CONFIG_DIRECTORY}/recc.conf
+        # 4. ~/.recc/recc.conf
+        # 5. #{etc}/recc.conf
+        ##
+
+        # Protocol version used for communicating with the REAPI server
+        reapi_version=2.2
+
+        # The URI of the server
+        server=unix://#{var}/recc/casd/casd.sock
+        # Instance name provided by the server
+        instance=recc-server
+
+        # Use cache-only mode, build anything not available in the cache locally
+        cache_only=1
+        # Upload results from the local builds to cache
+        cache_upload_local_build=1
+        # Upload failed results from the local builds to cache
+        cache_upload_failed_build=1
+
+        # Enable for link commands
+        link=1
+        # Use cache-only mode for link commands
+        link_cache_only=1
+
+        # Report all entries returned by the dependency command, even if they are absolute paths
+        deps_global_paths=1
+        # Do not rewrite absolute paths to be relative.
+        no_path_rewrite=1
+
+        # Do not retry
+        retry_limit=0
+
+        # Platform properties
+        REMOTE_PLATFORM_OS=#{OS.mac? ? "macos" : OS.kernel_name.downcase}
+        REMOTE_PLATFORM_ISA=#{Hardware::CPU.arch}
+      EOS
+    end
```

Comment:
```suggestion
    (buildpath/"recc.conf").write <<~EOS
      ##
      # For the list of configuration parameters please visit:
      #   https://buildgrid.gitlab.io/recc/configuration-variables.html
      #
      # The configuration file settings use lowercase letters without the RECC_ prefix
      # By default recc reads the configuration options from the following places,
      # applying settings bottom-up, with 1 being the last applied configuration.
      # If an option is set in multiple places, the one higher on this list will be the
      # effective one
      #
      # 1. Environment variables
      # 2. ${cwd}/recc/recc.conf
      # 3. {$RECC_CONFIG_DIRECTORY}/recc.conf
      # 4. ~/.recc/recc.conf
      # 5. #{etc}/recc.conf
      ##

      # Protocol version used for communicating with the REAPI server
      reapi_version=2.2

      # The URI of the server
      server=unix://#{var}/recc/casd/casd.sock
      # Instance name provided by the server
      instance=recc-server

      # Use cache-only mode, build anything not available in the cache locally
      cache_only=1
      # Upload results from the local builds to cache
      cache_upload_local_build=1
      # Upload failed results from the local builds to cache
      cache_upload_failed_build=1

      # Enable for link commands
      link=1
      # Use cache-only mode for link commands
      link_cache_only=1

      # Report all entries returned by the dependency command, even if they are absolute paths
      deps_global_paths=1
      # Do not rewrite absolute paths to be relative.
      no_path_rewrite=1

      # Do not retry
      retry_limit=0

      # Platform properties
      REMOTE_PLATFORM_OS=#{OS.mac? ? "macos" : OS.kernel_name.downcase}
      REMOTE_PLATFORM_ISA=#{Hardware::CPU.arch}
    EOS
    etc.install "recc.conf"
```
But to be honest I'm reluctant to have this in the formula -- it seems like something that will easily bitrot here.